tests/unit/machine/glm_unittest.cc

```diff
@@ -0,0 +1,88 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ * 
+ * Both the documentation and the code is heavily inspired by pyGLMnet.: https://github.com/glm-tools/pyglmnet/
+ *
+ * Author: Tej Sukhatme
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/machine/GLM.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/preprocessor/PruneVarSubMean.h>
+#include <shogun/preprocessor/NormOne.h>
+#include <shogun/mathematics/UniformRealDistribution.h>
+#include <shogun/mathematics/RandomNamespace.h>
+
+#include <random>
+
+using namespace shogun;
+
+//Generate the Data that N is greater than D
+void generate_train_data(SGMatrix<float64_t> &data, SGVector<float64_t> &lab)
+{
+
+	lab = SGVector<float64_t>({7.23514031, 7.23514031, 7.23514031, 7.23514031, 7.23514031});
+	data = SGMatrix<float64_t>({{ 0.71307143, -0.67054885, -0.24406853},
+								{-0.79774475, -1.65627891,  0.95675428},
+								{ 0.96709333,  1.81672959,  0.20911922},
+								{ 2.1912712,   0.23820139,  1.07501177},
+								{-0.58427793, -0.61855905,  1.27687684}});
+}
+
+void generate_test_data(SGMatrix<float64_t> &data, SGVector<float64_t> &lab)
+{
+
+	lab = SGVector<float64_t>({7.23514031, 7.23514031, 7.23514031, 7.23514031, 7.23514031});
+	data = SGMatrix<float64_t>({{ 1.26465769,  0.05451801, -0.21206714},
+								{-0.3447881,  -0.81339926,  1.636931  },
+								{ 0.3967461,  -1.6470009,   0.89995864},
+								{ 0.65379594,  1.08610417, -0.04911578},
+								{ 0.6573247,  -0.1306287,  -0.64715244}});
+}
+
+TEST(GLM, GLM_basic_test)
+{
+	SGMatrix<float64_t> Xtrain(3,5);
+	SGVector<float64_t> ytrain(5);
```

Comment:
here you are allocating memory, and then in the function you are just reallocating memory..

Reply:
Yes, I've done it the way you proposed earlier with the tuple.